### PR TITLE
field component and related refactoring

### PR DESCRIPTION
**Pull request by vincentv**

### CODE REVIEW

File: src/components/molecules/p6-field/p6-field.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    if (!isEmpty(this.input?.value)) {
+      this.checkValidity();
+    }
+  }
+
+  render(): JSX.Element {
+    const errorMessage =
+      this.errorMessage !== "" && !this.isReadOnly ? (
+        <p6-hint mode="danger" slot="hint">
```
Comment (vuongc) [inline src/components/molecules/p6-field/p6-field.tsx:41]: Why adding a `slot` attribute here ?

File: src/components/molecules/p6-field/p6-field.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    return (
+      <Host aria-disabled={this.isDisabled ? "true" : null}>
+        <label class="label" htmlFor={this.input?.name}>
+          <slot name="label" />
+          <slot />
+        </label>
+        {this.isReadOnly || this.isDisabled ? null : (
+          <div>
+            <slot name="hint" />
```
Comment (vuongc) [inline src/components/molecules/p6-field/p6-field.tsx:54]: Why adding a `name` attribute here ?

File: src/components/molecules/p6-field/utils.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { InputChild } from "./interface";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (vuongc) [inline src/components/molecules/p6-field/utils.ts:3]: Is this comment still necessary ? The InputChild isn't `any` in the _interface_ file.

File: src/components/atoms/p6-textarea/p6-textarea.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  Element,
+  h,
+  Host,
+  Method,
+  Prop,
+  State,
+} from "@stencil/core";
+import { cleanupValue } from "~utils/attribute";
+
+export type P6InputType =
```
Comment (jswale) [inline src/components/atoms/p6-textarea/p6-textarea.tsx:13]: dead code no?

File: src/components/molecules/p6-field/p6-field.tsx (hunk 2 above)
Comment (vincentv) [reply src/components/molecules/p6-field/p6-field.tsx:54]: to name the slot :thinking:

Comment (vuongc) [reply src/components/molecules/p6-field/p6-field.tsx]: it was more like why adding a slot instead of just displaying the component ?

**Pull request merged**